Nested project issues

When trying to compile System.IO.MemoryMappedFiles.sln in msbuild, the following error is shown:

> Solution file error MSB5009: Error parsing the nested project section in solution file.

Looking at the solution file, there indeed is a nested project section. If I understand it correctly, it's a remnant of 803180f3 and is indeed wrong.

Even with this, the project works fine in Visual Studio.

---

Similar error happens in System.Runtime.InteropServices.PInvoke.sln:

> Solution file error MSB5023: Error parsing the nested project section in solution file. A project with the GUID "{A824F4CD-935B-4496-A1B2-C3664936DA7B}" is listed as being nested under project "{B754B947-C1E1-4FFE-B03B-CEAC451913DC}", but does not exist in the solution.

This time, it seems the error is that System.Runtime.InteropServices.PInvoke.Tests.csproj is missing from the solution.

This also does not cause an error in VS.

---

I'll submit pull request for both shortly.

--- a/src/System.IO.MemoryMappedFiles/System.IO.MemoryMappedFiles.sln
+++ b/src/System.IO.MemoryMappedFiles/System.IO.MemoryMappedFiles.sln
@@ -35,8 +35,4 @@ Global
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
 	EndGlobalSection
-	GlobalSection(NestedProjects) = preSolution
-		{9D6F6254-B5A3-40FF-8925-68AA8D1CE933} = {10E9121A-7350-4AB7-BB05-A543BE1EB19B}
-		{16EE5522-F387-4C9E-9EF2-B5134B043F37} = {10E9121A-7350-4AB7-BB05-A543BE1EB19B}
-	EndGlobalSection
 EndGlobal

--- a/src/System.Runtime.InteropServices.PInvoke/System.Runtime.InteropServices.PInvoke.sln
+++ b/src/System.Runtime.InteropServices.PInvoke/System.Runtime.InteropServices.PInvoke.sln
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 14
-VisualStudioVersion = 14.0.23107.0
+VisualStudioVersion = 14.0.24720.0
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Runtime.InteropServices.PInvoke", "src\System.Runtime.InteropServices.PInvoke.csproj", "{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}"
 EndProject
@@ -12,6 +12,8 @@ EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "ref", "ref", "{E8CF0B11-55CA-408D-A489-5727F3ED6A17}"
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Runtime.InteropServices.PInvoke", "ref\System.Runtime.InteropServices.PInvoke.csproj", "{B17014F1-D902-417F-89B0-271204695831}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Runtime.InteropServices.PInvoke.Tests", "tests\System.Runtime.InteropServices.PInvoke.Tests.csproj", "{A824F4CD-935B-4496-A1B2-C3664936DA7B}"
 EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
@@ -31,10 +33,10 @@ Global
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
 		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Debug|Any CPU.ActiveCfg = net46_Debug|Any CPU
 		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Debug|Any CPU.Build.0 = net46_Debug|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Linux_Debug|Any CPU.ActiveCfg = Linux_Debug|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Linux_Debug|Any CPU.Build.0 = Linux_Debug|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Linux_Release|Any CPU.ActiveCfg = Linux_Release|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Linux_Release|Any CPU.Build.0 = Linux_Release|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Linux_Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Linux_Debug|Any CPU.Build.0 = Debug|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Linux_Release|Any CPU.ActiveCfg = Release|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Linux_Release|Any CPU.Build.0 = Release|Any CPU
 		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.net46_Debug|Any CPU.ActiveCfg = net46_Debug|Any CPU
 		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.net46_Debug|Any CPU.Build.0 = net46_Debug|Any CPU
 		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.net46_Release|Any CPU.ActiveCfg = net46_Release|Any CPU
@@ -43,16 +45,16 @@ Global
 		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.netcore50aot_Debug|Any CPU.Build.0 = netcore50aot_Debug|Any CPU
 		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.netcore50aot_Release|Any CPU.ActiveCfg = netcore50aot_Release|Any CPU
 		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.netcore50aot_Release|Any CPU.Build.0 = netcore50aot_Release|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.OSX_Debug|Any CPU.ActiveCfg = OSX_Debug|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.OSX_Debug|Any CPU.Build.0 = OSX_Debug|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.OSX_Release|Any CPU.ActiveCfg = OSX_Release|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.OSX_Release|Any CPU.Build.0 = OSX_Release|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.OSX_Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.OSX_Debug|Any CPU.Build.0 = Debug|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.OSX_Release|Any CPU.ActiveCfg = Release|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.OSX_Release|Any CPU.Build.0 = Release|Any CPU
 		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Release|Any CPU.ActiveCfg = net46_Release|Any CPU
 		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Release|Any CPU.Build.0 = net46_Release|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Windows_Debug|Any CPU.ActiveCfg = Windows_Debug|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Windows_Debug|Any CPU.Build.0 = Windows_Debug|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Windows_Release|Any CPU.ActiveCfg = Windows_Release|Any CPU
-		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Windows_Release|Any CPU.Build.0 = Windows_Release|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Windows_Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Windows_Debug|Any CPU.Build.0 = Debug|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Windows_Release|Any CPU.ActiveCfg = Release|Any CPU
+		{EC6AA4D9-B3E8-4CCA-8AB1-8BBFD89266AE}.Windows_Release|Any CPU.Build.0 = Release|Any CPU
 		{B17014F1-D902-417F-89B0-271204695831}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{B17014F1-D902-417F-89B0-271204695831}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{B17014F1-D902-417F-89B0-271204695831}.Linux_Debug|Any CPU.ActiveCfg = Debug|Any CPU

